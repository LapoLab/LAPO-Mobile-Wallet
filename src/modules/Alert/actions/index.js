// @flow
import { createActions } from 'redux-actions';

export const { showAlert, hideAlert } = createActions({
  SHOW_ALERT: (title = '', content = '', buttons = [], config) => ({
    title,
    content,
    buttons,
    config,
  }),
  HIDE_ALERT: (data = '') => data,
});
