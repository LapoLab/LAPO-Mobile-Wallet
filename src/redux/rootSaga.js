import { all } from 'redux-saga/effects';
import commonsSagas from '../commons/sagas';
import recoveryWordsSagas from '../modules/RecoveryWords/sagas';
import sendSagas from '../modules/Send/sagas';
import paperWordsConfirm from '../modules/PaperWordsConfirm/sagas';
import dashboard from '../modules/Dashboard/sagas';
import settings from '../modules/Settings/sagas';
import paperWords from '../modules/PaperWords/sagas';
import transactionsDetails from '../modules/TransactionDetails/sagas';
import chooseContact from '../modules/ChooseContact/sagas';
import splashScreen from '../modules/SplashScreen/sagas';
import switchContacts from '../modules/SwitchContacts/sagas';
import receive from '../modules/Receive/sagas';

// eslint-disable-next-line
const getListeners = (...args) => args.reduce((acc, nextArg) => [...acc, ...Object.values(nextArg).map(func => func())], []);

export default function* rootSaga() {
  yield all(
    getListeners(
      commonsSagas,
      recoveryWordsSagas,
      sendSagas,
      paperWordsConfirm,
      dashboard,
      settings,
      paperWords,
      transactionsDetails,
      chooseContact,
      splashScreen,
      switchContacts,
      receive,
    ),
  );
}
