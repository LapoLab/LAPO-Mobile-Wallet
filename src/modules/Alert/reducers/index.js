import { handleActions } from 'redux-actions';
import { showAlert, hideAlert } from '../actions';

export const initialAlertState = {
  title: '',
  content: '',
  buttons: [],
  isShow: false,
};

// reducer
const reducer = handleActions(
  {
    [showAlert](
      state,
      {
        payload: { title, content, buttons },
      },
    ) {
      return {
        ...state,
        title,
        content,
        isShow: true,
        buttons,
      };
    },
    [hideAlert](
      state,
    ) {
      return {
        ...state,
        isShow: false,
      };
    },
  },
  initialAlertState,
);

export default reducer;
