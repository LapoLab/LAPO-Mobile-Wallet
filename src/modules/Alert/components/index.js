/**
 * @flow
 */

import React, { Component } from 'react';
import {
  Text, View, Modal, TouchableHighlight,
} from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  DialogFooter,
  DialogButton,
  DialogContent,
  DialogTitle,
  // $FlowFixMe
} from 'react-native-popup-dialog';
import * as actions from '../actions';
import styles from './styles';

type Props = {
  title: string,
  content: string,
  buttons: Array<{ text: string, onPress: Function }>,
  isShow: boolean,
  hideAlert: Function,
};
export class AlertModal extends Component<Props> {
  handleClose = () => {
    const { hideAlert, buttons } = this.props;
    hideAlert();
    if (buttons && buttons.length === 1 && buttons[0].onPress) {
      buttons[0].onPress();
    }
  };

  render() {
    const {
      title, content, isShow, buttons,
    } = this.props;

    return (
      <Modal transparent animationType="fade" visible={isShow} onRequestClose={this.handleClose}>
        <TouchableHighlight
          style={styles.GlobalContainer}
          onPress={this.handleClose}
          underlayColor="transparent"
        >
          <View style={styles.container}>
            <View style={styles.content}>
              <DialogTitle
                title={title}
                style={{
                  padding: 20,
                }}
                hasTitleBar={false}
                align="left"
              />
              <DialogContent style={styles.content}>
                <Text style={styles.contentText}>{content}</Text>
              </DialogContent>
              <DialogFooter>
                {buttons.map(item => (
                  <DialogButton
                    key={item.text}
                    text={item.text}
                    style={{
                      padding: 0,
                      borderRadius: 12,
                    }}
                    textStyle={{
                      color: 'gray',
                      fontSize: 14,
                    }}
                    onPress={item.onPress || this.handleClose}
                  />
                ))}
              </DialogFooter>
            </View>
          </View>
        </TouchableHighlight>
      </Modal>
    );
  }
}

const mapStateToProps = state => ({
  title: state.alert.title,
  content: state.alert.content,
  buttons: state.alert.buttons,
  isShow: state.alert.isShow,
});

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    hideAlert: actions.hideAlert,
  },
  dispatch,
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AlertModal);
