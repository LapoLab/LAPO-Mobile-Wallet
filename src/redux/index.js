import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import rootReducer from './rootReducer';
import rootSagas from './rootSaga';

const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
  blacklist: [
    'modals',
    'dashboard',
    'paperWords',
    'chooseContact',
    'send',
    'transactionsDetails',
    'alert',
    'pinsRecovery',
  ],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const sagaMiddleware = createSagaMiddleware();
// eslint-disable-next-line no-undef
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSagas);

export const persistor = persistStore(store);
