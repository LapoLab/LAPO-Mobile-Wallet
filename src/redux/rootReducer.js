import { combineReducers } from 'redux';

import wallet from '../commons/reducers/wallet';
import pins from '../commons/reducers/pins';
import pinsRecovery from '../commons/reducers/pinsRecovery';
import modals from '../commons/reducers/modals';
import settings from '../commons/reducers/settings';
import notify from '../modules/Notify/reducers';
import paperConfirm from '../modules/PaperWordsConfirm/reducer';
import dashboard from '../modules/Dashboard/reducers';
import paperWords from '../modules/PaperWords/reducers';
import chooseContact from '../modules/ChooseContact/reducers';
import send from '../modules/Send/reducers';
import transactionsDetails from '../modules/TransactionDetails/reducer';
import alert from '../modules/Alert/reducers';

export default combineReducers({
  wallet,
  pins,
  pinsRecovery,
  notify,
  modals,
  settings,
  paperConfirm,
  dashboard,
  paperWords,
  chooseContact,
  send,
  transactionsDetails,
  alert,
});
