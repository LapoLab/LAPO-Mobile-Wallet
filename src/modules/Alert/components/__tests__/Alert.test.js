// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../redux';
import Alert from '../index';

describe('Alert page', () => {
  test('renders snapshots', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Alert />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
