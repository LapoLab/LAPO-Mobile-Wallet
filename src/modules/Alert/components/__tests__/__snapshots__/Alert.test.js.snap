// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert page renders snapshots 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.45)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 12,
          "elevation": 6,
          "justifyContent": "center",
          "minHeight": 180,
          "shadowColor": "rgb(0,0,0)",
          "shadowOffset": Object {
            "height": 4,
            "width": 2,
          },
          "shadowOpacity": 0.34,
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "padding": 24,
                "paddingLeft": 18,
                "paddingRight": 18,
              },
              Object {
                "alignItems": "flex-start",
              },
              null,
              Object {
                "padding": 20,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#151822",
                  "fontFamily": "sans-serif-medium",
                  "fontSize": 19,
                  "fontWeight": "400",
                },
                null,
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "paddingBottom": 24,
                "paddingLeft": 18,
                "paddingRight": 18,
              },
              Object {
                "flex": 1,
                "height": "100%",
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "borderColor": "#CCD0D5",
                "borderTopWidth": 0.2,
              },
              null,
            ]
          }
        />
      </View>
    </View>
  </View>
</Modal>
`;
